File: core/matchers/src/main/scala/au/com/dius/pact/core/matchers/util/CollectionUtils.scala
package au.com.dius.pact.core.matchers.util

import scala.collection.JavaConversions

object CollectionUtils {

  def toOptionalList(list: java.util.List[String]): Option[List[String]] = {
    if (list == null) {
      None
    } else {
      Some(JavaConversions.collectionAsScalaIterable(list).toList)
    }
  }
}

File: core/matchers/src/main/kotlin/au/com/dius/pact/core/matchers/util/CollectionUtils.kt
package au.com.dius.pact.core.matchers.util

import java.util.Collections.nCopies

fun <E> List<E>.padTo(size: Int, item: E): List<E> {
  return when {
    size < this.size -> subList(fromIndex = 0, toIndex = size)
    size > this.size -> this + nCopies(size - this.size, item)
    else -> this
  }
}
